ui/selectors/transactions.js

```diff
@@ -251,7 +251,16 @@ export const nonceSortedTransactionsSelector = createSelector(
         txReceipt,
       } = transaction;
 
-      if (typeof nonce === 'undefined' || type === TransactionType.incoming) {
+      let undefinedNonceOrIncomingTransaction =
+        typeof nonce === 'undefined' || type === TransactionType.incoming;
+
+      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+      // Don't group transactions by nonce if they are custodial TXes
+      undefinedNonceOrIncomingTransaction =
+        undefinedNonceOrIncomingTransaction || transaction.custodyId;
```

Comment:
It might be best to cast the custodyId to a boolean to help with the inference that we're just checking for its existence, rather than doing anything with its value

Reply:
Good idea, will do 👍🏼

Reply:
Also, lets rename 'undefinedNonceOrIncomingTransaction' to something like 'shouldNotBeGrouped' and add a documentation above the variable declaration detailing the 3 specific cases we have for it.